exporters/geneva/include/opentelemetry/exporters/geneva/metrics/socket_tools.h

```diff
@@ -252,6 +254,12 @@ struct SocketAddr {
       // Max length of Unix domain filename is up to 108 chars
       strncpy_s(m_data_un.sun_path, sizeof(m_data_un.sun_path),
                 unix_domain_path, sizeof(m_data_un.sun_path));
+      // special handling for abstract socket - they should be
+      // prefixed with '@' character.
+      if (unix_domain_path[0] == '@') {
+        m_data_un.sun_path[0] = '\0';
```

Comment:
Why do we need set `sun_path[0]` to NULL?

Reply:
The abstract socket starts with a NULL char as the first character. We ask for the socket name in the exporter configuration and prefix it with '@' (the printable char for abstract socket) before passing it to socket_tools.h. And then replace '@' with '\0' here.

https://man7.org/linux/man-pages/man7/unix.7.html

```
abstract: an abstract socket address is distinguished (from a
          pathname socket) by the fact that sun_path[0] is a null byte
          ('\0'). 
```

NULL char in `char *` can mess with the string processing, so it is replaced in the end.

Reply:
Why is it necessary to use the `@` symbol to represent `NULL` character in the exporter configuration, rather than assuming that it will be automatically prepended to the user's input path? Additionally, must we continue to support non-abstract UDS?

Reply:
So, these are the connection strings we support as export options:

   `Account={MetricAccount};NameSpace={MetricNamespace}`     - ETW
  `Endpoint=unix://{UDS Path};Account={MetricAccount};Namespace={MetricNamespace}`   - UDS (abstract/non-abstract)
If `{UDS path}` is prefixed with `@` character, this would mean the USD socket is of abstract type. The UDS non-abstract file path can never have this prefix, so it should be safe to do so.

The other option without prefix `@` character would be:

   `Account={MetricAccount};NameSpace={MetricNamespace}`     - ETW
  `Endpoint=unix://{UDS Path};Account={MetricAccount};Namespace={MetricNamespace}`   - UDS non-abstract
  `Endpoint=abs://{UDS Path};Account={MetricAccount};Namespace={MetricNamespace}`  - UDS abstract
And then pass the abstract flag to socket_tools.h which will create the appropriate socket based on this flag.

It was just the implementation choice :)

Also on Linux, `lsof` and `netstat` commands will display an abstract socket prefixed with this character, so the character is familiar one and logically makes sense:

```bash
$ lsof -U

upowerd   1604       root    5u  unix 0xffff88005af5f400      0t0 18631 type=STREAM
colord    1614     colord   10u  unix 0xffff880034d3f400      0t0 18170 type=STREAM
systemd   2009       root   13u  unix 0xffff88005a293000      0t0 21213 /run/user/0/systemd/notify type=DGRAM
systemd   2009       root   14u  unix 0xffff88005a293c00      0t0 21214 /run/user/0/systemd/private type=STREAM
upstart   1525    lightdm    7u  unix 0xffff880034b99800      0t0 17301 @/com/ubuntu/upstart-session/111/1525 type=STREAM

```